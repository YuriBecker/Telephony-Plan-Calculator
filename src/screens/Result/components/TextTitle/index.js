import React from 'react';
import { StyleSheet } from 'react-native';
import { Text } from '@ui-kitten/components';
import PropTypes from 'prop-types';
import colors from '../../../../styles/colors';

const styles = StyleSheet.create({
  title: {
    textAlign: 'center',
    color: colors.primary,
  },
});

const index = ({ children, category, style }) => {
  return (
    <Text category={category} style={[styles.title, style]}>
      {children}
    </Text>
  );
};

export default index;

index.propTypes = {
  children: PropTypes.node.isRequired,
  category: PropTypes.string,
};

index.defaultProps = {
  category: 'h2',
};
