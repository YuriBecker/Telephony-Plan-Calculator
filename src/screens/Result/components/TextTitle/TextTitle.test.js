import React from 'react';
import { shallow } from 'enzyme';

import TextTitle from './index';

describe('Testing TextTitle component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TextTitle> Some Title</TextTitle>);
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());
});
