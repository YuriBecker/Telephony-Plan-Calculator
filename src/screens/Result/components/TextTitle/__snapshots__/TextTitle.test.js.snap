// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TextTitle component should render correctly 1`] = `
<TextComponent
  category="h2"
  style={
    Array [
      Object {
        "color": "#1db954",
        "textAlign": "center",
      },
      undefined,
    ]
  }
>
   Some Title
</TextComponent>
`;
